src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -4574,6 +4612,27 @@ mesh::matset::verify(const Node &matset,
         }
     }
 
+    if(mat_map_is_optional && matset.has_child("material_map"))
+    {
+        log::optional(info, protocol, "includes material_map");
+        res &= verify_matset_material_map(protocol,matset,info);
+
+        // for cases where vfs are an object, we expect the material_map names to match
+        if(matset.has_child("volume_fractions") &&
+           matset["volume_fractions"].dtype().is_object())
+        {
+            const std::vector<std::string> &vf_mats  = matset["volume_fractions"].child_names();
+            const std::vector<std::string> &map_mats = matset["material_map"].child_names();
+            const std::set<std::string> vf_matset(vf_mats.begin(), vf_mats.end());
+            const std::set<std::string> map_matset(map_mats.begin(), map_mats.end());
+            if(vf_matset != map_matset)
```

Comment:
I think this should probably check that `vf_matset` is a subset of `map_matset` instead of forcing them to be equal. It's a slim and out-there use case, but I could imagine a user having one `material_map` spread across all `matset` entries, and in that case it may have extra material values that aren't relevant to some particular entries.

Reply:
done, also added a test case